### ISSUE TITLE
Check `match` keyword arguments

### ISSUE THREAD

**Issue by aDotInTheVoid** [microsoft/pyright#9121]
**Is your feature request related to a problem? Please describe.**

The following code is buggy, as it pattern matches on a field that doesn't exist.

```python
# pyright: strict

import dataclasses


@dataclasses.dataclass
class Foo:
    some_name: str


def match_on_foo(f: Foo) -> int:
    match f:
        case Foo(other_name="hello"):
            return 1
        case _:
            return 0
```

However, pyright doesn't detect this:

```
% pyright ./mcve.py
0 errors, 0 warnings, 0 informations
```

**Describe the solution you’d like**

Pyright should warn/error when matching on an unknown field.

**VS Code extension or command-line**

pyright 1.1.382, on CLI

**Comment by erictraut** [comment]
The code sample is legitimate because there could be a subclass of `Foo` that introduces an attribute named `other_name`. Even if there were no such subclass, the code will still run without generating an exception. For those reasons, I don't think it would be appropriate for a type checker to flag this as an error. It would lead to false positive errors. Here's an example.

```python
from dataclasses import dataclass

@dataclass
class Foo:
    some_name: str

@dataclass
class Bar(Foo):
    other_name: str

def match_on_foo(f: Foo):
    match f:
        case Foo(other_name="hello"):
            print("hello")
        case _:
            pass

match_on_foo(Bar(some_name="", other_name="hello"))
```

**Comment by aDotInTheVoid** [comment]
Thanks for the detailed explanation. I marked my classes as `@typing.final`, and then got an error as I hoped.
